modules/navier_stokes/src/bcs/INSADMomentumNoBCBC.C

```diff
@@ -33,16 +38,25 @@ INSADMomentumNoBCBC::INSADMomentumNoBCBC(const InputParameters & parameters)
   : ADVectorIntegratedBC(parameters),
     _p(adCoupledValue("p")),
     _integrate_p_by_parts(getParam<bool>("integrate_p_by_parts")),
-    _mu(getADMaterialProperty<Real>("mu_name"))
+    _mu(getADMaterialProperty<Real>("mu_name")),
+    _form(getParam<MooseEnum>("viscous_form"))
 {
+  auto & obj_tracker = const_cast<INSADObjectTracker &>(
+      _fe_problem.getUserObject<INSADObjectTracker>("ins_ad_object_tracker"));
+  obj_tracker.setViscousForm(_form);
+  obj_tracker.setIntegratePByParts(_integrate_p_by_parts);
 }
 
 ADReal
 INSADMomentumNoBCBC::computeQpResidual()
 {
   // The viscous term
-  // -mu * (grad(u).n) * test
-  ADReal residual = -_mu[_qp] * (_grad_u[_qp] * _normals[_qp]) * _test[_i][_qp];
+  ADReal residual;
+  if (_form == "laplace")
+    residual = -_mu[_qp] * (_grad_u[_qp] * _normals[_qp]) * _test[_i][_qp];
+  else
+    residual =
+        -_mu[_qp] * ((_grad_u[_qp] + _grad_u[_qp].transpose()) * _normals[_qp]) * _test[_i][_qp];
```

Comment:
`((_grad_u[_qp] + _grad_u[_qp].transpose()) * _normals[_qp])` needes to be multiplied by `0.5`? When considering `grad_u[_qp] ` "equals" `0.5*(_grad_u[_qp] + _grad_u[_qp].transpose())`

I did not do the math on this, and so I do not know for sure.

Reply:
You may already cancel out somewhere else.

Reply:
Look at equation 8 of our INS paper :smile:

Reply:
Oh, I see! Thanks